File: app/src/main/java/io/github/thang86/weathers/base/MvpPresenter.java
package io.github.thang86.weathers.base;

/**
 * ThangTX2
 */
public interface MvpPresenter<V extends BaseView> {

    /**
     * attcht the view
     *
     * @param view
     */
    void attachView(V view);

    /**
     * detach the view
     */
    void detachView();
}

File: app/src/main/java/io/github/thang86/weathers/core/MvpPresenter.kt
package io.github.thang86.weathers.core

interface MvpPresenter<V : BaseView> {
    /**
     * attcht the view
     *
     * @param view
     */
    fun attachView(view: V)

    /**
     * detach the view
     */
    fun detachView()
}